janus/message-base.md

```diff
@@ -0,0 +1,306 @@
+# SSI Base Message Format
+
+> *Note: This is a rough document, and contains inconsistancies, missing sections, etc. These will iron out as work on this document progresses.*
+
+The role of the Self-Soverign Identity Message Format is to act as a foundation for the message types that will power the SSI messaging ecosystem.
+This format is intended to be as simple as possible, providing just enough structure to facilitate compatible communication.
+
+Goals:
+* Broad compatibility across SSI ecosystem
+* Allow for innovation in message types
+* Support communication from clients with commonly available libraries
+
+## Base Envelope
+
+The base message format is a JWT Payload. All JWT standard attributes are reserved. Additional attributes to messages are included based on the type specified. (See Message Types.)
+
+Example:
+```JSON
+{
+    "@type": "example.com/basemessage",
+    "iss": "did:exa:11111111111",
+    "aud": "did:exa:22222222222",
+    "iat": "timestamp",
+}
+```
+### Applicable JWT attributes
+
+* **"iss"** (Issuer) - from DID
+* **"sub"** (Subject)
```

Comment:
JWT presents itself as "a compact, URL-safe means of representing claims to be transferred between two parties." For that reason, it makes sense that it has these common "claim names". However, these claim names assume the signing algorithm is a very simple one, like RSA, DSA, or ECDSA. More advanced ZK signing algorithms are clearly not contemplated. The JWT spec has one paragraph on privacy, and it has two options (1) encryption, and (2) don't include information. Correlation privacy isn't mentioned.

If we say we're using JWT, it will be understood that we're using it for claims exchange, which we can't. We can use JWS and JWE as "dumb" wrappers, but the main purpose of JWT will be lost on Sovrin. We could take the portions of JWT that don't hurt correlation privacy and use just those, but I suspect others in the SSI ecosystem will want to use "iss", "sub", etc.